src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -933,7 +933,7 @@ internal bool CalculateUseSiteDiagnostic(ref DiagnosticInfo result)
 
             // Check return type, custom modifiers, parameters
             if (DeriveUseSiteDiagnosticFromType(ref result, this.ReturnTypeWithAnnotations,
-                                                MethodKind == MethodKind.PropertySet ?
+                                                IsInitOnly ?
```

Comment:
Nit: in the future, it would be great to have this bug fix as either a separate commit or a separate PR, as it doesn't really seem related to VB consuming such properties.

Reply:
> Nit: in the future, it would be great to have this bug fix as either a separate commit or a separate PR, as it doesn't really seem related to VB consuming such properties.

I think the change is related to the feature I am working on. And I don't think it makes sense to separate this change and have compilers behave inconsistently.

---
In reply to: [557614115](https://github.com/dotnet/roslyn/pull/50414#discussion_r557614115) [](ancestors = 557614115)